test/kaniko_task_test.go

```diff
@@ -165,27 +132,137 @@ func TestKanikoTaskRun(t *testing.T) {
+}
+
+func getRegistryService(namespace string) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      "registry",
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{{
+				Port: 5000,
+			}},
+			Selector: map[string]string{
+				"app": "registry",
+			},
+		},
+	}
+}
+
+func getGitResource(namespace string) *v1alpha1.PipelineResource {
+	return tb.PipelineResource(kanikoGitResourceName, namespace, tb.PipelineResourceSpec(
+		v1alpha1.PipelineResourceTypeGit,
+		tb.PipelineResourceSpecParam("Url", "https://github.com/GoogleContainerTools/kaniko"),
+		tb.PipelineResourceSpecParam("Revision", revision),
+	))
+}
+
+func getImageResource(namespace, repo string) *v1alpha1.PipelineResource {
+	return tb.PipelineResource(kanikoImageResourceName, namespace, tb.PipelineResourceSpec(
+		v1alpha1.PipelineResourceTypeImage,
+		tb.PipelineResourceSpecParam("url", repo),
+	))
+}
+
+func getTask(repo, namespace string) *v1alpha1.Task {
+	taskSpecOps := []tb.TaskSpecOp{
+		tb.TaskInputs(tb.InputsResource("gitsource", v1alpha1.PipelineResourceTypeGit)),
+		tb.TaskOutputs(tb.OutputsResource("builtImage", v1alpha1.PipelineResourceTypeImage)),
+	}
+	stepOps := []tb.StepOp{
+		tb.StepArgs(
+			"--dockerfile=/workspace/gitsource/integration/dockerfiles/Dockerfile_test_label",
+			fmt.Sprintf("--destination=%s", repo),
+			"--context=/workspace/gitsource",
+			"--oci-layout-path=/workspace/output/builtImage",
+			"--insecure",
+			"--insecure-pull",
+			"--insecure-registry=registry"+namespace+":5000/",
+		),
+	}
+	step := tb.Step("kaniko", "gcr.io/kaniko-project/executor:v0.13.0", stepOps...)
+	taskSpecOps = append(taskSpecOps, step)
+	sidecar := tb.Sidecar("registry", "registry")
+	taskSpecOps = append(taskSpecOps, sidecar)
+
+	return tb.Task(kanikoTaskName, namespace, tb.TaskSpec(taskSpecOps...))
+}
+
+func getTaskRun(namespace string) *v1alpha1.TaskRun {
+	return tb.TaskRun(kanikoTaskRunName, namespace, tb.TaskRunSpec(
+		tb.TaskRunTaskRef(kanikoTaskName),
+		tb.TaskRunTimeout(2*time.Minute),
+		tb.TaskRunInputs(tb.TaskRunInputsResource("gitsource", tb.TaskResourceBindingRef(kanikoGitResourceName))),
+		tb.TaskRunOutputs(tb.TaskRunOutputsResource("builtImage", tb.TaskResourceBindingRef(kanikoImageResourceName))),
+	))
+}
+
+func getRemoteDigest(t *testing.T, c *clients, namespace, image string) (string, error) {
+	t.Helper()
+	podName := "skopeo-jq"
+	if _, err := c.KubeClient.Kube.CoreV1().Pods(namespace).Create(&corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      podName,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Name:    "skopeo",
+				Image:   "quay.io/rhpipeline/skopeo:alpine",
+				Command: []string{"/bin/sh", "-c"},
+				Args:    []string{"apk add --no-cache jq 1>/dev/null; skopeo inspect --tls-verify=false docker://" + image + ":latest| jq '.Digest'"},
```

Comment:
I'm gonna push some images to `plumbing` to use it instead of this hack :angel:

Reply:
gcr.io/tekton-releases/dogfooding/skopeo is now available